Common/include/containers/container_decorators.hpp

```diff
@@ -205,3 +205,60 @@ struct C3DDummyMiddleView
     return data(i,k);
   }
 };
+
+/*--- Helper functions to allocate containers of containers. ---*/
+
+/*!
+ * \brief Allocate a vector of varying-size vectors and initialize with some value.
+ * \param[in] M - the first index is >=0 and < M
+ * \param[in] N - the second index is >=0 and < N[first index]
+ * \param[in,out] X - the vector of vectors
+ * \param[in] val - the value for initialization, default is 0
+ * \tparam IndexVector - type of N
+ * \tparam VectorOfVector - type of X
+ */
+template<class IndexVector, class VectorOfVector, class Scalar = int>
+inline void AllocVectorOfVectors(size_t M, const IndexVector& N, VectorOfVector& X, Scalar val = 0) {
+  X.resize(M);
+  for(size_t i = 0; i < M; ++i){
+    X[i].resize(N[i]);
+    for (auto& x : X[i]) x = val;
```

Comment:
I changed it a bit to try to make it compatible with more types. If I broke anything I'll fix it.
Thanks for all the cleanup.